src/core/Schemas.ts

```diff
@@ -33,8 +33,9 @@ export type Intent =
   | BuildUnitIntent
   | EmbargoIntent
   | QuickChatIntent
-  | MoveWarshipIntent
-  | MarkDisconnectedIntent;
+  | MarkDisconnectedIntent
+  | UpgradeStructureIntent
+  | MoveWarshipIntent;
```

Comment:
Let's try to minimize the reordering with:
```suggestion
  | MoveWarshipIntent
  | MarkDisconnectedIntent
  | UpgradeStructureIntent
  ;
```